pootle/static/js/editor/components/SuggestionFeedbackForm.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import FormElement from 'components/FormElement';
+import FormMixin from 'mixins/FormMixin';
+
+import { decodeEntities } from '../utils';
+
+export const SuggestionFeedBackForm = React.createClass({
+
+  propTypes: {
+    suggId: React.PropTypes.number.isRequired,
+    suggText: React.PropTypes.string.isRequired,
+    editor: React.PropTypes.object.isRequired,
```

Comment:
It's highly discouraged to pass around heavy objects like the entire editor module being passed here. Generally speaking this is bad because the component needs to make too many assumptions about the shape of the data being passed, and it also makes it difficult to implement lifecycle methods like `shouldComponentUpdate()`. Passing the values/functions needed by the component explicitly via props is the suggested approach, and while it could feel more verbose, it's also easier to refactor and lint.